src/definitions/x509/crl.rs

```diff
@@ -0,0 +1,320 @@
+use asn1_rs::{FromDer, Sequence};
+use const_oid::{AssociatedOid, ObjectIdentifier};
+use der::{Any, Decode, SliceReader};
+use signature::Verifier;
+use thiserror::Error;
+use x509_cert::{
+    crl::{CertificateList, RevokedCert, TbsCertList},
+    ext::pkix::{
+        name::{DistributionPointName, GeneralName},
+        CrlDistributionPoints, CrlReason,
+    },
+    spki::AlgorithmIdentifierOwned,
+    TbsCertificate,
+};
+
+/// All CRL parsing and revocation errors
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Cert was revoked: Reason:{0:?}")]
+    CertRevoked(Option<CrlReason>),
+
+    #[error("Unable to parse CRL component: {0} {1}")]
+    ParsingCrlComponent(&'static str, der::Error),
+    #[error("Distrbution point malformed: {0}")]
+    DistributionPointMalformed(&'static str),
+    #[error("Unable to fetch CRL: {0}")]
+    FetchingCrl(reqwest::Error),
+    #[error("Unable to reach distribution point")]
+    ReachingDistributionPoint,
+    #[error("Issuer mismatch between cert and CRL")]
+    IssuerMismatchBetweenCertAndCrl,
+    #[error("Unknown signature algorithm: {0:?}")]
+    UnknownSignatureAlgorithm(Box<AlgorithmIdentifierOwned>),
+    #[error("Signature type mismatch: {0:?} {0:?}")]
+    SignatureTypeMismatch(Box<AlgorithmIdentifierOwned>, Box<AlgorithmIdentifierOwned>),
+    #[error("Unable to extract CRL sequence signature")]
+    ExtractCrlSequenceSignature(asn1_rs::Err<asn1_rs::Error>),
+    #[error("Missing curve")]
+    MissingCurve,
+    #[error("Unknown curve extension: {0}")]
+    UnknownCurveExt(ObjectIdentifier),
+    #[error("Missing public key")]
+    MissingPublicKey,
+    #[error("Signature check of CRL failed")]
+    SignatureCheckOfCrlFailed,
+    #[error("Invalid public key format")]
+    InvalidPublicKeyFormat,
+    #[error("Missing signature")]
+    MissingSignature,
+    #[error("Signature in wrong format")]
+    SignatureInWrongFormat,
+}
+
+pub const OID_EC_CURVE_P256: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.10045.3.1.7");
```

Comment:
This library should burden itself with defining mappings. It should instead rely on the `pkcs8` crate with the `AssociatedOid` trait. Then the verification function can accept an object that implements both `DecodePublicKey` and the `Verifier` trait from the `signature` trait -- or something along these lines, maybe `DynAssociatedAlgorithmIdentifier` will be useful. Then it's on the consumer to pick the "crypto backend". And it allows to build an object that can support multiple types of keys. 

A "default" verifier can be provided by this library, but feature gated.